src/intervals/trigonometric.jl

```diff
@@ -1,9 +1,21 @@
 # This file is part of the IntervalArithmetic.jl package; MIT licensed
 
-half_pi(::Type{T}) where T = pi_interval(T) / 2
-half_pi(x::T) where T<:AbstractFloat = half_pi(T)
 
-two_pi(::Type{T})  where T = pi_interval(T) * 2
+# hard code this for efficiency:
+const one_over_half_pi_interval = inv(0.5 * pi_interval(Float64))
+
+"""
+Multiply an interval by a positive constant.
+For efficiency, does not check that the constant is positive.
+"""
+multiply_by_positive_constant(α, x::Interval) = @round(α*x.lo, α*x.hi)
```

Comment:
Perhaps instead of defining `multiply_by_positive_constant` we should rather define the methods `*(a::T, x::Interval)` and  `*(x::Interval, a::T)` (in `arithmetic.jl`), where `T<:Real` (or maybe `<:Number`), or even unspecified, as it is here. Currently we use promotion to an interval and rounding, which may slow down things.

Question: Below you use this function for to define `half_pi(::Type{Float64})`, but not for other types; can you explain why?

Reply:
I use it for `two_pi` as well.

Actually the `half_pi` function is no longer used, I believe.

We may well want to define explicit methods for `*(constant, Interval)`. I've opened an issue: https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/120

Reply:
That was actually the suggestion.

Reply:
Right, but I'd prefer to leave it for a separate PR.
Also, if you actually know that the constant is positive then this specialised version is even slightly quicker.